test/CMakeLists.txt

```diff
@@ -17,3 +17,9 @@ if(TARGET ${PROJECT_NAME}_utest)
   target_link_libraries(${PROJECT_NAME}_utest ${Boost_LIBRARIES} ${class_loader_LIBRARIES})
   add_dependencies(${PROJECT_NAME}_utest ${PROJECT_NAME}_TestPlugins1 ${PROJECT_NAME}_TestPlugins2)
 endif()
+
+catkin_add_gtest(${PROJECT_NAME}_unique_ptr_test unique_ptr_test.cpp)
+if(TARGET ${PROJECT_NAME}_unique_ptr_test)
```

Comment:
Good point, I'm used to working with GCC 6 which did compile the test (it defaults to `-std=gnu++14`). 

Test updated: removed the `#ifdef` in the test and instead made the compilation conditional in the `CMakeLists.txt` with the flags added if supported by the compiler.